sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -0,0 +1,2244 @@
+    };
+
+    static Azure::Core::Http::Request FileSystemDeleteCreateRequest(std::string url, const FileSystemDeleteOptions& fileSystemDeleteOptions)
+    {
+      Azure::Core::Http::Request request(Azure::Core::Http::HttpMethod::Delete, std::move(url));
+      request.AddQueryParameter(k_QueryFileSystemResource, "filesystem");
+      if (!fileSystemDeleteOptions.ClientRequestId.empty())
+      {
+        request.AddHeader(k_HeaderClientRequestId, fileSystemDeleteOptions.ClientRequestId);
+      } // TODO: This should be changed to null check when Optional<T> is ready.
+      if (!std::to_string(fileSystemDeleteOptions.Timeout).empty())
+      {
+        request.AddQueryParameter(k_QueryTimeout, std::to_string(fileSystemDeleteOptions.Timeout));
+      }
+      request.AddHeader(k_HeaderApiVersionParameter, fileSystemDeleteOptions.ApiVersionParameter);
+      if (!fileSystemDeleteOptions.IfModifiedSince.empty())
+      {
+        request.AddHeader(k_HeaderIfModifiedSince, fileSystemDeleteOptions.IfModifiedSince);
+      }
+      if (!fileSystemDeleteOptions.IfUnmodifiedSince.empty())
+      {
+        request.AddHeader(k_HeaderIfUnmodifiedSince, fileSystemDeleteOptions.IfUnmodifiedSince);
+      }
+      return request;
+    }
+
+    struct FileSystemDeleteResponse
+    {
+      std::string RequestId;
+      std::string Version;
+      std::string Date;
+    };
+
+    static FileSystemDeleteResponse FileSystemDeleteParseResponse(/*const*/ std::unique_ptr<Azure::Core::Http::Response>& responsePtr)
+    {
+      /* const */ auto& response = *responsePtr;
+      if (response.GetStatusCode() == Azure::Core::Http::HttpStatusCode::Accepted)
+      {
+        // Accepted
+        FileSystemDeleteResponse result;
+        result.RequestId = response.GetHeaders().at(k_HeaderXMsRequestId);
+        result.Version = response.GetHeaders().at(k_HeaderXMsVersion);
+        result.Date = response.GetHeaders().at(k_HeaderDate);
+        return result;
+      }
+      else
+      {
+        std::cout << "Error occured:\n" << std::string(response.GetBodyBuffer().begin(), response.GetBodyBuffer().end()) << std::endl;
+        // throw StorageError::CreateFromXml(XmlWrapper::CreateFromBodyBuffer(response.GetBodyBuffer()));
+      }
+    }
+
+    static FileSystemDeleteResponse FileSystemDelete(std::string url, std::shared_ptr<Azure::Core::Http::HttpTransport> transport, Azure::Core::Context& context, const FileSystemDeleteOptions& fileSystemDeleteOptions)
+    {
+      // TODO: Pipeline will be added when available.
+      return FileSystemDeleteParseResponse(transport->Send(context, FileSystemDeleteCreateRequest(std::move(url), fileSystemDeleteOptions)));
+    }
+
+    struct FileSystemListPathsOptions
+    {
+      std::string ClientRequestId = std::string(); // Provides a client-generated, opaque value with a 1 KB character limit that is recorded in the analytics logs when storage analytics logging is enabled.
```

Comment:
Still don't think it's right to define some variable in one way and some in another way, just to imply whether it's optional.

I think we should escalate and discuss about this in the meeting with Core team.

Reply:
Agreed, I'll add this to the OneNote page and ignore this at the moment.